src/boot/SerializeEscaped.js

```diff
@@ -0,0 +1,155 @@
+/* @flow strict-local */
+import type { Refs, Replacer, Reviver } from 'remotedev-serialize';
+import * as Serialize from 'remotedev-serialize';
+import Immutable from 'immutable';
+
+import objectEntries from '../utils/objectEntries';
+
+// The syntax used in template literals is unhelpfully similar to
+// RegExp pattern syntax, so let's avoid template literals, and use
+// simple string concatenation with `+` instead.
+/* eslint-disable prefer-template */
+
+/**
+ * A special string used by `remotedev-serialize`.
+ *
+ * A value at this key is used to mark if and how a piece of data
+ * should be revived into its non-serializable form.
+ *
+ * Always use this constant, in custom replacers/revivers, for
+ * enforced consistency -- don't make new string literals. That way,
+ * there is no doubt that what needs to be escaped will be escaped.
+ *
+ * This should also match the string that `remotedev-serialize` uses
+ * in its built-in replacers and revivers.
+ */
+export const SERIALIZED_TYPE_FIELD_NAME: '__serializedType__' = '__serializedType__';
+
+/**
+ * The single character to use in the escape strategy, as a prefix.
+ *
+ * This is not meant to be changed. We declare this `const` variable
+ * to clarify the escape/unescape patterns and enforce consistency
+ * between them.
+ *
+ * If you need to change this at all, be sure to write a migration. If
+ * you need to make it more than one character, also rename the
+ * variable and modify the escape/unescape patterns accordingly.
+ *
+ * Exported only for tests.
+ */
+export const ESCAPE_CHARACTER: '!' = '!';
+
+type StringTransform = (oldString: string) => string;
+
+/**
+ * Escape a string matching /^(!*)__serializedType__$/ as /^!\1__serializedType__$/
+ *
+ * Exported only for tests.
+ */
+export const escape: StringTransform = (key: string) =>
```

Comment:
This comment applies to the `escape` function and the `unescape` function:

Here, I've used a string concatenated with help from the ESCAPE_CHARACTER and SERIALIZED_TYPE_FIELD_NAME constants to construct a single RegExp. I suppose it would have been possible to break it up into multiple RegExps, but it isn't a very long or complicated string.

Also (and this didn't apply when we were doing the ZulipVersion parsing), we need to make an absolute yes-or-no decision on any input string. To me, that suggests that we should strongly consider a single RegExp (even if the concatenated string we'd pass to it is very long or complicated, which, here, it's not). I mentioned this preference [here](https://chat.zulip.org/#narrow/stream/65-checkins/topic/rohitt/near/827349).